salvo/src/lib/benchmark/base_benchmark.py

```diff
@@ -0,0 +1,68 @@
+"""
```

Comment:
(optional, food for thought) How would you feel about using pytype annotations in code in this repository? They tend to help a lot by being able to statically discover most common programming errors in Python.

Let me know if this is something you would be willing to explore, I can provide some hints.

Some background:
https://github.com/google/styleguide/blob/gh-pages/pyguide.md#221-type-annotated-code

Reply:
I can investigate and add this later once I address the other comments.

Reply:
(optional) We certainly don't have to add pytype annotations in this PR (and we aren't going to block this PR on this comment).

I did find the pytype annotations very helpful in preventing bugs and shortening the debug cycle on Python code, so if you are interested and inclined towards using them here I will be happy to work with you and help you figure out how to configure the build system to make them work.

Let's address this in a slack conversation as you have suggested.